diff --git a/numpy/_core/src/multiarray/descriptor.c b/numpy/_core/src/multiarray/descriptor.c
--- a/numpy/_core/src/multiarray/descriptor.c
+++ b/numpy/_core/src/multiarray/descriptor.c
@@ -2525,6 +2525,7 @@ arraydescr_new(PyTypeObject *subtype,
                 PyErr_NoMemory();
                 return NULL;
             }
+            PyObject_Init((PyObject *)descr, subtype);
             Py_XINCREF(DType->scalar_type);
             descr->typeobj = DType->scalar_type;
             descr->type_num = DType->type_num;
